controller/engine_controller.go

```diff
@@ -1199,22 +1204,44 @@ func restoreBackup(log logrus.FieldLogger, engine *longhorn.Engine, rsMap map[st
 		return errors.Wrapf(err, "cannot get backup target config for backup restoration of engine %v", engine.Name)
 	}
 
+	backoff := flowcontrol.NewBackOff(retryRestoreEntryTime, retryRestoreMaxInterval)
+	startTime := time.Now()
+	// Register the name with a backkofentry
+	backoff.Next(engine.Name, startTime)
 	if cliAPIVersion < engineapi.CLIVersionFour {
 		// For compatible engines, `LastRestoredBackup` is required to indicate if the restore is incremental restore
 		log.Infof("Prepare to restore backup, backup target: %v, backup volume: %v, requested restored backup name: %v, last restored backup name: %v",
 			backupTargetClient.URL, engine.Spec.BackupVolume, engine.Spec.RequestedBackupRestore, engine.Status.LastRestoredBackup)
-		if err = engineClientProxy.BackupRestore(engine, backupTargetClient.URL, engine.Spec.RequestedBackupRestore, engine.Spec.BackupVolume, engine.Status.LastRestoredBackup, backupTargetClient.Credential); err != nil {
-			if extraErr := handleRestoreErrorForCompatibleEngine(log, engine, rsMap, err); extraErr != nil {
-				return extraErr
+		for {
```

Comment:
You cannot use a for loop (with sleep) here since it will block the sync between the engine CR and the engine process.
The whole monitor sync function will be executed every several seconds. You can directly use the backoff function here: If it's in the backoff window, directly skip this part.

Reply:
OK, changed, add `restoreBackoff` into EngineMonitor to record the backoff.
Do you think `restoreBackup` could be an EngineMonitor method?

Reply:
Yes you can do it.

Reply:
Changed, thanks